### ISSUE TITLE
Adjusting training script for early stopping callback

### ISSUE THREAD

**Issue by CoCoNuTeK** [amazon-science/chronos-forecasting#107]
Hello there,
so i made these adjustments to the train.py
added to my config and the script main():
```
eval_data_paths: str,
patience: int = 20,
frequency: str = "5min"

eval_datasets = [
    Filter(
        partial(
            has_enough_observations,
            min_length=min_past + prediction_length,
            max_missing_prop=max_missing_prop,
        ),
        FileDataset(path=Path(data_path), freq=frequency),
    )
    for data_path in eval_data_paths
]

eval_dataset = ChronosDataset(
        datasets=eval_datasets,
        probabilities=probability,
        tokenizer=chronos_config.create_tokenizer(),
        context_length=context_length,
        prediction_length=prediction_length,
        min_past=min_past,
        mode="validation",
    )
```

inside the training_args i put 
```
load_best_model_at_end=True,
metric_for_best_model="eval_loss", 
greater_is_better=False,  
```

and inside the Trainer i put 
```
logging_strategy="epoch",
save_strategy="epoch",
eval_strategy="epoch",
eval_dataset=eval_dataset
callbacks=[EarlyStoppingCallback(early_stopping_patience=patience)]
```

i tried finetuing the 'small' model i put 1K epochs, however i am not quite sure it had any effect as the training ran for full 1K epochs and from serialization i got checkpoint-1000 and also checkpoint-final as outputs of that run

Here i would like to ask, if i set number of epochs to 1K the checkpoint-final contains the best model over the whole training session and the checkpoint-1000 are weights after all the 1K epochs i assume right?

**Comment by lostella** [comment]
@CoCoNuTeK assuming the callback is from HF transformers, this is a question more for https://huggingface.co/docs/transformers/en/main_classes/callback

I guess you're right about checkpoint-1000, not sure about checkpoint-final, but my guess is it contains the model from the step where it decided to stop training.

**Comment by CoCoNuTeK** [comment]
Nevermind i was using steps instead of epochs i will need to tinker a bit more with the train.py, just to help me out a bit
the __iter__(self) -> Iterator inside of the ChronosDataset does that need to return all the data at once?? in the 
```
        return {
            "input_ids": input_ids.squeeze(0),
            "attention_mask": attention_mask.squeeze(0),
            "labels": labels.squeeze(0),
        }
```
format??

I wont have missing data so the attention_mask should be all 1s right and then what is the input_ids/labels??

if i have 10 sequences of length lets say 500 the input_ids should be what shape? and same for the labels

thank you very much

**Comment by CoCoNuTeK** [comment]
Okay I solved it.
